### DISCUSSION TITLE
Enable splitting on `addProtocol`

### DISCUSSION THREAD

**Discussion by helloimalastair** [maplibre/maplibre-gl-js#2247] [Q&A]
Is there any way to allow `addProtocol` to be imported directly from `maplibre-gl`, rather than having to import the entire `maplibre` object? Trying to reduce my code bundle size...

So like, instead of
```js
import maplibre from "maplibre-gl";
maplibre.addProtocol(...);
```
doing this:
```js
import { addProtocol } from "maplibre-gl";
addProtocol(...);
```

**Comment by HarelM** [comment]
This is a global definition related to configuration, I don't think you can simply import that part unfortunately.
If you think this can be improved somehow please let us know (i.e. in a PR).

**Reply by helloimalastair** [reply]
Isn't there a way to pipe a custom protocol directly into a `maplibre.Map`? I'm able to import `Map` separately in a way that allows it to be separated from the rest of the code, so it seems odd that configuration requires me to import the entire package...

**Reply by HarelM** [reply]
The configuration is for the workers as well, not just the map.
I'm theory you might be able to achieve the same with service workers, I'm not sure.

**Reply by helloimalastair** [reply]
Ok, now I'm wondering, is it actually supported to pre-bundle the `maplibre-gl` library, or should it always be run through a CDN? Because AFAIK, the code bundled doesn't include any service workers...

**Reply by HarelM** [reply]
The code is bundled with the initialization of web workers, but to listen and alter the network communication you can use service workers, which are not part of maplibre
